### PR TITLE
Remove redundant marine ctests

### LINKED ISSUE #306: Clean-up/Speed-up the soca ctests
Remove the ctests that are mpi dependent when not building with the g-w.
This is the current-ish list of tests:
```
   Test  #1: test_gdasapp_check_python_norms
  Test  #2: test_gdasapp_check_yaml_keys
  Test  #3: test_gdasapp_check_valid_yaml
  Test  #4: test_gdasapp_jedi_increment_to_fv3
  Test  #5: test_gdasapp_genYAML_prep
  Test  #6: test_gdasapp_genYAML_run
  Test  #7: test_gdasapp_convert_ewok_yaml
  Test  #8: test_gdasapp_setup_cycled_exp
  Test  #9: test_gdasapp_convert_bufr_temp_dbuoy
  Test #10: test_gdasapp_convert_bufr_salt_dbuoy
  Test #11: test_gdasapp_convert_bufr_temp_mbuoyb
  Test #12: test_gdasapp_convert_bufr_salt_mbuoyb
  Test #13: test_gdasapp_convert_bufr_tesacprof
  Test #14: test_gdasapp_convert_bufr_trkobprof
  Test #15: test_gdasapp_convert_bufr_sfcships
  Test #16: test_gdasapp_convert_bufr_sfcshipsu
  Test #17: test_gdasapp_soca_obsdb
  Test #18: test_gdasapp_soca_ana_prep
  Test #19: test_gdasapp_soca_ana_bmat
  Test #20: test_gdasapp_soca_ana_run
  Test #21: test_gdasapp_soca_ana_bmat_vrfy
  Test #22: test_gdasapp_soca_run_clean
  Test #23: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP
  Test #24: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT
  Test #25: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN
  Test #26: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_POST
  Test #27: test_gdasapp_land_create_ens
  Test #28: test_gdasapp_land_imsproc
  Test #29: test_gdasapp_land_apply_jediincr
  Test #30: test_gdasapp_land_letkfoi_snowda
  Test #31: test_gdasapp_convert_bufr_adpsfc
  Test #32: test_gdasapp_convert_gsi_satbias
  Test #33: test_gdasapp_store_gsi_satbias
  Test #34: test_gdasapp_setup_atm_cycled_exp
  Test #35: test_gdasapp_atm_jjob_var_prep
  Test #36: test_gdasapp_atm_jjob_var_run
  Test #37: test_gdasapp_atm_jjob_var_post
  Test #38: test_gdasapp_atm_jjob_ens_prep
  Test #39: test_gdasapp_atm_jjob_ens_run
  Test #40: test_gdasapp_atm_jjob_ens_post
  Test #41: test_gdasapp_aero_gen_3dvar_yaml
```
Tests 19-21 are SLOW and redundant with 24-26, and cover less code. **Remove**

### PR DESCRIPTION
**Pull request by guillaumevernieres**
Fixes the current `soca` related failures.

But also:
- fixes #306

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
92%, is that an **A**? 
```
Tests: 92% tests passed, 3 tests failed out of 39
	1355 - test_gdasapp_land_letkfoi_snowda (Failed)
	1361 - test_gdasapp_atm_jjob_var_run (Failed)
	1364 - test_gdasapp_atm_jjob_ens_run (Failed)
Tests: see output at /work2/noaa/stmp/cmartin/CI/GDASApp/workflow/PR/335/global-workflow/sorc/gdas.cd/build/log.ctest
```
@RussTreadon-NOAA has 1361 & 1364 sorted out in a different PR.
@jiaruidong2017 is working on 1355.

File: parm/soca/obs/obs_list.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 observers:
 - !INC ${OBS_YAML_DIR}/adt_j3.yaml
-- !INC ${OBS_YAML_DIR}/salt_profile_fnmoc.yaml
+#- !INC ${OBS_YAML_DIR}/salt_profile_fnmoc.yaml
```
Comment (RussTreadon-NOAA) [inline parm/soca/obs/obs_list.yaml:3]: This change was committed at 53b99c6 with the comment `removing broken ufo `.  Just curious, are there plans to fix the ufo and reactivate this line?

**Review by RussTreadon-NOAA — APPROVED** [review]
Changes look good.   Curiosity question is just that - I'm curious.   It's not a blocker. 
Approve.

**Comment by AndrewEichmann-NOAA** [comment]
Philosophical musing: I get that the jjob tests test these as well, insofar as the scripts in /scripts are dependencies, but I wonder it it's better to keep them in the spirit of making unit testing as atomic as possible.

**Review by CoryMartin-NOAA — APPROVED** [review]

File: parm/soca/obs/obs_list.yaml (hunk 1 above)
Comment (guillaumevernieres) [reply parm/soca/obs/obs_list.yaml:3]: yes, and I'm not sure the failure is related to the recent changes or even this repo. I added an issue a few days back.

**Comment by guillaumevernieres** [comment]
> Philosophical musing: I get that the jjob tests test these as well, insofar as the scripts in /scripts are dependencies, but I wonder it it's better to keep them in the spirit of making unit testing as atomic as possible.

That's a point that I was struggling with too @AndrewEichmann-NOAA , but more codes means more time spent maintaining it. That chunk of ci was clunky and regularly breaking. I don't think it was worth keeping. What we could do is start putting together real unit tests, at a lower level than the script.

**Comment by guillaumevernieres** [comment]
@CoryMartin-NOAA @RussTreadon-NOAA and @AndrewEichmann-NOAA , it needs a re-review ...

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**